From 02276aa6da403d815ad0c69b5cd0277bdea84933 Mon Sep 17 00:00:00 2001
From: Zhao Wei Liew <zhaoweiliew@gmail.com>
Date: Sun, 17 Jul 2016 11:53:42 +0800
Subject: [PATCH 071/173] Snap: Always allow 100% JPEG quality to be set

Previously, we were checking if the selected JPEG quality was 100%
and not setting the JPEG quality if some conditions were met.

There isn't any history nor explanation on why we weren't setting
the JPEG quality to 100%, so just remove the redundant check.

Change-Id: Ie0ab555ad35942bd115751fd10e4f4c11fa7a76d
---
 src/com/android/camera/PhotoModule.java | 28 ++++++++++---------------
 1 file changed, 11 insertions(+), 17 deletions(-)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index c69ff774d..0ef76b5e4 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -3247,26 +3247,20 @@ public class PhotoModule
         String jpegQuality = mPreferences.getString(
                 CameraSettings.KEY_JPEG_QUALITY,
                 mActivity.getString(R.string.pref_camera_jpegquality_default));
-        //mUnsupportedJpegQuality = false;
         Size pic_size = mParameters.getPictureSize();
         if (pic_size == null) {
             Log.e(TAG, "error getPictureSize: size is null");
-        }
-        else{
-            if("100".equals(jpegQuality) && (pic_size.width >= 3200)){
-                //mUnsupportedJpegQuality = true;
-            }else {
-                mParameters.setJpegQuality(JpegEncodingQualityMappings.getQualityNumber(jpegQuality));
-                int jpegFileSize = estimateJpegFileSize(pic_size, jpegQuality);
-                if (jpegFileSize != mJpegFileSizeEstimation) {
-                    mJpegFileSizeEstimation = jpegFileSize;
-                    mHandler.post(new Runnable() {
-                        @Override
-                        public void run() {
-                            updateRemainingPhotos();
-                        }
-                    });
-                }
+        } else {
+            mParameters.setJpegQuality(JpegEncodingQualityMappings.getQualityNumber(jpegQuality));
+            int jpegFileSize = estimateJpegFileSize(pic_size, jpegQuality);
+            if (jpegFileSize != mJpegFileSizeEstimation) {
+                mJpegFileSizeEstimation = jpegFileSize;
+                mHandler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        updateRemainingPhotos();
+                    }
+                });
             }
         }
 
-- 
2.17.1

